zeus/utils/framework.py

```diff
@@ -42,18 +42,46 @@ def torch_is_available():
         return False
 
 
-def cuda_sync(device: int | None = None) -> None:
-    """Synchronize CPU and CUDA.
+@lru_cache(maxsize=1)
+def jax_is_available():
+    """Check if JAX is available."""
+    try:
+        import jax
+
+        assert jax.devices("gpu"), "JAX is available but does not have CUDA support."
+        MODULE_CACHE["jax"] = jax
+        logger.info("JAX with CUDA support is available.")
+        return True
+    except ImportError:
+        logger.info("JAX is not available")
+        return False
+
+
+def cuda_sync(device: int | None = None, backend: str = "torch") -> None:
+    """Synchronize CPU with CUDA.
 
     Note: `cupy.cuda.Device.synchronize` may be a good choice to make
           CUDA device synchronization more general. Haven't tested it yet.
 
     Args:
         device: The device to synchronize.
+        backend: The backend framework. Defaults to `torch`
     """
-    if torch_is_available():
+    if backend == "torch" and torch_is_available():
         torch = MODULE_CACHE["torch"]
-        torch.cuda.synchronize(device)
-        return
 
-    raise RuntimeError("No frameworks are available.")
+        def synchronize_cuda_fn(device):
+            torch.cuda.synchronize(device)
+
+    elif backend == "jax" and jax_is_available():
+        jax = MODULE_CACHE["jax"]
+
+        def synchronize_cuda_fn(device):
+            (
+                jax.device_put(0.0, device=jax.devices("gpu")[device]) + 0
+            ).block_until_ready()
+
+    else:
+        raise RuntimeError("No framework is available.")
+
+    synchronize_cuda_fn(device)
```

Comment:
A couple comments:
1. If the user passed in a specific `backend` like `"torch"` but PyTorch could not be imported, I think that should be an error. To preserve the stack trace (In case the user's PyTorch env was screwed up and `import torch` was not working as intended, that'll help them to debug), I would suggest:
   -  `*_is_available` accepts a new parameter `ensure_available: bool = False`
   - ```python
     try:
         ...
     except ImportError as e:
         logger.info("PyTorch is not available")
         if ensure_available:
             raise RuntimeError("Failed to import PyTorch") from e
         return False
      ```

1. I find it unnecessary to add an extra layer of function call with `synchronize_cuda_fn`. I think calling the synchronization primitive inside the if/elif clause and immediately returning (like before the change) is better.

1. The input argument `device` can be `None`, in which case `torch.cuda.synchronize` calls `cudaSynchronize` on the *current device*, which can be set and queried with `torch.cuda.set_device` and `torch.cuda.current_device`.
  For Jax, I did some searching (I'm not really a Jax user) and found that its called *default device* in Jax can be set temporarily using the context manager `jax.default_device`, and `jax.device_put` with `device=None` will use the default device. So I think the following could work; what do you think?
   ```python
   jax.device_put(0.0, device=None if device is None else jax.devices("gpu")[devices])
   ```